docs/auto-instrumentation-install.md

```diff
@@ -0,0 +1,278 @@
+
+This Instrumentation object is a spec to configure what instrumentation libraries to use for instrumentation. An
+Instrumentation object must be available to the target pod for auto-instrumentation to function. Here is an example
+of a Instrumentation yaml file and how to install it.
+
+<details open>
+<summary>Expand for sample splunk-instrumentation.yaml</summary>
+
+```yaml
+# splunk-instrumentation.yaml
+apiVersion: opentelemetry.io/v1alpha1
+kind: Instrumentation
+metadata:
+  name: splunk-instrumentation
+spec:
+  exporter:
+    endpoint: http://$(SPLUNK_OTEL_AGENT):4317
+  propagators:
+    - tracecontext
+    - baggage
+    - b3
+  dotnet:
+    env:
+      - name: SPLUNK_OTEL_AGENT
+        valueFrom:
+          fieldRef:
+            apiVersion: v1
+            fieldPath: status.hostIP
+      - name: OTEL_EXPORTER_OTLP_ENDPOINT
+        value: http://$(SPLUNK_OTEL_AGENT):4317
+      - name: OTEL_TRACES_EXPORTER
+        value: otlp
+```
+
+</details>
+
+```
+# Install
+kubectl apply -f splunk-instrumentation.yaml
+# Check the current deployed values
+kubectl get otelint -o yaml
+```
+
+### 4. Verify all the OpenTelemetry resources (collector, operator, webhook, instrumentation) are deployed successfully
+
+<details open>
+<summary>Expand for sample output to verify against</summary>
+
+```
+kubectl  get pods -n monitoring
+# NAME                                                          READY
+# STATUS    RESTARTS   AGE
+# splunk-otel-collector-agent-9ccgn                             2/2     Running   0          3m
+# splunk-otel-collector-agent-ft4xc                             2/2     Running   0          3m
+# splunk-otel-collector-k8s-cluster-receiver-56f7c9cf5b-mgsbj   1/1     Running   0          3m
+# splunk-otel-collector-operator-6dffc898df-5jjkp               2/2     Running   0          3m
+
+kubectl get mutatingwebhookconfiguration.admissionregistration.k8s.io -n monitoring
+# NAME                                      WEBHOOKS   AGE
+# cert-manager-webhook                      1          8m
+# splunk-otel-collector-operator-mutation   3          2m
+
+kubectl get otelinst -n spring-petclinic
+# NAME                          AGE   ENDPOINT
+# splunk-instrumentation        3m   http://$(SPLUNK_OTEL_AGENT):4317
+```
+
+</details>
+
+### 5. Instrument application by setting an annotation
+
+An _inject instrumentation_ annotation can be added to the following.
+- Namespace: All pods within that namespace will be instrumented.
+- Pod Spec Objects: PodSpec objects that are available as part of Deployment,
+  Statefulset, or other resources can be annotated.
+  following values.
+- "true" - inject and Instrumentation resource from the namespace.
+- "my-instrumentation" - name of Instrumentation CR instance in the current
+  namespace.
+- "my-other-namespace/my-instrumentation" - name and namespace of.
+  Instrumentation CR instance in another namespace.
+- "false" - do not inject.
```

Comment:
These parts are pretty unclear to me as a user, with the header saying: "An _inject instrumentation_ annotation can be added to the following."

Reply:
This is a direct copy of the upstream documentation, this has been a pain point for new users. I do think these docs could be improved. I need to search the Operator code in depth to actually add any more details about this, I'd likely add these new docs to the upstream project as well.

I'll see what I can do here.